admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/service/impl/SyncConfigDetailsServiceImpl.java

```diff
@@ -47,227 +46,310 @@
+                            + ExceptionUtils.buildMessage(ex.getMessage(), ex.getCause()));
+        }
+
+        try {
+            publicKeyResponseMapped = objectMapper.readValue(publicKeyResponseEntity.getBody(),
+                    new TypeReference<ResponseWrapper<PublicKeyResponse<String>>>() {
+                    });
+
+            publicKeyResponseMapped.getResponse().setProfile(environment.getActiveProfiles()[0]);
+            LOGGER.debug("Public key fetched successfully for applicationId: {}", applicationId);
+            return publicKeyResponseMapped.getResponse();
+        } catch (IOException | NullPointerException e) {
+            LOGGER.error("Failed to parse public key response: {}", e.getMessage());
+            throw new SyncDataServiceException(SyncConfigDetailsErrorCode.SYNC_IO_EXCEPTION.getErrorCode(),
+                    SyncConfigDetailsErrorCode.SYNC_IO_EXCEPTION.getErrorMessage(), e);
+        }
+    }
+
+    /**
+     * Retrieves and encrypts configuration details for a machine.
+     *
+     * @param keyIndex the machine key index or name
+     * @return the {@link ConfigDto} containing encrypted global and registration configurations
+     * @throws RequestException if no machine is found
+     * @throws SyncDataServiceException if fetching or encryption fails
+     */
+    @Override
+    public ConfigDto getConfigDetails(String keyIndex) {
+        Machine machine = validateKeyIndexAndGetMachine(keyIndex);
+        LOGGER.debug("Fetching config details for machine keyIndex: {}, machine: {}",
+                keyIndex.replaceAll("[\n\r]", "_"), machine.getName());
+
+        JSONObject config = new JSONObject();
+        JSONObject globalConfig = new JSONObject();
+        JSONObject regConfig = parsePropertiesString(configServerClient.fetch(regCenterfileName));
+        //This is not completely removed only for backward compatibility, all the configs will be part of registrationConfiguration
+        config.put("globalConfiguration", getEncryptedData(globalConfig, machine));
+        config.put("registrationConfiguration", getEncryptedData(regConfig, machine));
+
+        ConfigDto configDto = new ConfigDto();
+        configDto.setConfigDetail(config);
+        LOGGER.debug("Get ConfigDetails() {} completed", keyIndex.replaceAll("[\n\r]", "_"));
+        return configDto;
+    }
+
+    /**
+     * Retrieves a script, optionally encrypted, for a machine.
+     *
+     * @param scriptName the name of the script
+     * @param keyIndex   the machine key index or name
+     * @return a {@link ResponseEntity} containing the script content
+     * @throws RequestException if no machine is found
+     * @throws SyncInvalidArgumentException if the script name is invalid
+     * @throws SyncDataServiceException if fetching or encryption fails
+     */
+    @Override
+    public ResponseEntity getScript(String scriptName, String keyIndex) throws Exception {
+        validateScriptName(scriptName);
```

Comment:
Can we sanitise script name once instead repeating it again and again?

Reply:
taken care